README.md

```diff
@@ -9,6 +9,100 @@
+    )
+)
+reporter.info('This is something important') // prints to stdout in cyan: This is something important
+reporter.error('An error occurred') // prints to stderr in red: [ERROR] An error occured
+reporter.warn('Something non-critical went wrong') // prints to stderr in yellow: [WARNING] Something non-critical went wrong
+reporter.debug('This will only be printed when --verbose is passed') // prints to stdout if --verbose in dim gray: [DEBUG] This will only be printed when --verbose is passed
+
+// Use prompt to ask the user for input - it is a thin wrapper around `inquirer.prompt`, see https://github.com/SBoudrias/Inquirer.js
+const answers = await prompt([
+    {
+        type: 'list',
+        name: 'theme',
+        message: 'What do you want to do?',
+        choices: [
+            'Order a pizza',
+            'Make a reservation',
+            new inquirer.Separator(),
+            'Ask for opening hours',
+            {
+                name: 'Contact support',
+                disabled: 'Unavailable at this time',
+            },
+            'Talk to the receptionist',
+        ],
+    },
+    {
+        type: 'list',
+        name: 'size',
+        message: 'What size do you need?',
+        choices: ['Jumbo', 'Large', 'Standard', 'Medium', 'Small', 'Micro'],
+        filter: function (val) {
+            return val.toLowerCase()
+        },
+        when: function (answers) {
+            return answers.theme === 'Order a pizza'
+        },
+    },
+])
+if (answers.theme === 'Order a pizza') {
+    reporter.info(`Ordering a ${answers.size} pizza`)
+} else {
+    reporter.info(`Ok, let's ${answers.theme}`)
+}
+
+// Exec a function in the local shell environment, pass some arguments, set the current working directory, and capture the output as a string
+
+const textContents = await exec({
+    cmd: 'cat',
+    args: ['./test.txt'],
+    cwd: './myDirectory',
+    captureOut: true,
+})
+```
+
+There's a lot more you can do with these utilities - check the source code in the `./libs` folder or the many CLI module examples at [dhis2/cli](https://github.com/dhis2/cli) for inspiration.
+
+## How to use the d2 Cache
+
+Any CLI command instantiated with `@dhis2/cli-helpers-engine` will have an injected `getCache` argument. This is a powerful, simple cache engine for downloading, caching, and accessing files in the global `d2` cache.
+
+The most handy method of the `d2` Cache is `get`, which allows the developer to fetch a remote URL, optionally extract it if the download is a zip file, and store it in the cache. If a copy already exists in the cache, the download is skipped (unless `force: true` is specified):
+
+```js
+const url =
+    'https://spreadsheets.google.com/feeds/list/1Fd-vBoJPjp5wdCyJc7d_LOJPOg5uqdzVa3Eq5-VFR-g/2/public/values?alt=json'
+
+// This will download a json file listing all countries which implement DHIS2 and store it in
+cache.get(url, 'dhis2-in-action-countries.json', { force: true })
```

Comment:
I think we should add the import and initialization statement to make this a complete example